src/relation/mod.rs

```diff
@@ -723,6 +773,59 @@ impl JoinConstraint {
             JoinConstraint::None => JoinConstraint::None,
         }
     }
+
+    pub fn has_unique_constraint(
+        &self,
+        left_schema: &Schema,
+        right_schema: &Schema,
+    ) -> (bool, bool) {
+        match self {
+            JoinConstraint::On(x) => match x {
+                Expr::Function(f) if f.function() == function::Function::Eq => {
+                    let is_unique_fields = Hierarchy::from_iter(
```

Comment:
Rename (F2) -> fields_with_unique_or_primary_key_constraint

Reply:
A a general rule an identifier starting with `is` or `has` is either a function or method returning a bool or the result of such function (or similar expression).